x11-libs/cairo/patches/cairo-1.14.12.patchset

```diff
@@ -0,0 +1,92 @@
+From 888e4094afccd2016aa72828b7d49263ee80d8db Mon Sep 17 00:00:00 2001
+From: Jerome Duval <jerome.duval@gmail.com>
+Date: Fri, 3 Oct 2014 18:33:08 +0000
+Subject: Haiku patch
+
+
+diff --git a/configure.ac b/configure.ac
+index 2ce1959..c04c05c 100644
+--- a/configure.ac
++++ b/configure.ac
+@@ -287,6 +287,11 @@ CAIRO_ENABLE_SURFACE_BACKEND(beos, BeOS/Zeta, no, [
+       AC_CHECK_LIB(be,main,beos_LIBS="$beos_LIBS -lbe")
+       AC_CHECK_LIB(zeta,main,beos_LIBS="$beos_LIBS -lzeta")
+       ;;
++    *-*-haiku)
++      beos_LIBS=""
++      AC_CHECK_LIB(be,main,beos_LIBS="$beos_LIBS -lbe")
++      AC_CHECK_LIB(network,main,beos_LIBS="$beos_LIBS -lnetwork")
++      ;;
+     *)
+       use_beos="no (requires a BeOS platform)"
+       ;;
+-- 
+2.10.2
+
+
+From e3e4f468cc12266d121a2ddb99808a090b15e157 Mon Sep 17 00:00:00 2001
+From: Adrien Destugues <pulkomandy@gmail.com>
+Date: Mon, 6 Oct 2014 18:07:36 +0200
+Subject: Add missing include fenv.h.
+
+
+diff --git a/test/cairo-test.c b/test/cairo-test.c
+index a351b01..474f34c 100644
+--- a/test/cairo-test.c
++++ b/test/cairo-test.c
+@@ -40,6 +40,7 @@
+ #include <unistd.h>
+ #endif
+ #include <errno.h>
++#include <fenv.h>
+ #include <string.h>
+ #if HAVE_FCFINI
+ #include <fontconfig/fontconfig.h>
+-- 
+2.10.2
+
+
+From 66cf163733ee56380b0525b918b7969849ed0229 Mon Sep 17 00:00:00 2001
+From: Jerome Duval <jerome.duval@gmail.com>
+Date: Fri, 21 Nov 2014 22:11:40 +0000
+Subject: link tests against libnetwork.
```

Comment:
And how is it exactly better? How can it be upstreamed if it resides in the recipe?

Reply:
This patch just add `-lnetwork` unconditionally, and of course won't be accepted by upstream at all. If one wanted to upstream it, I would recommend fixing the build system so that `-lnetwork` is added automatically on Haiku.

IMO, unless the patch is improved so that it could be upstreamed, we should just drop it to reduce maintainance cost.